src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -395,19 +395,10 @@ void AppHost::_revokeWindowCallbacks()
 
 // Method Description:
 // - Called every time when the active tab's title changes. We'll also fire off
-//   a window message so we can update the window's title on the main thread,
-//   though we'll only do so if the settings are configured for that.
-// Arguments:
-// - sender: unused
-// - newTitle: the string to use as the new window title
-// Return Value:
-// - <none>
-void AppHost::_AppTitleChanged(const winrt::Windows::Foundation::IInspectable& /*sender*/, winrt::hstring newTitle)
+//   a window message so we can update the window's title on the main thread.
+void AppHost::_AppTitleChanged(const winrt::Windows::Foundation::IInspectable& /*sender*/, const winrt::Windows::Foundation::IInspectable& /*args*/)
 {
-    if (_windowLogic.GetShowTitleInTitlebar())

```

Comment:
removing these checks across the codebase consolidates the behavior in TerminalPage::Title.